pkg/aws/v1alpha3/machinedeployment/mutate_machinedeployment.go

```diff
@@ -179,13 +200,52 @@ func (m *Mutator) MutateInfraRef(machineDeployment capiv1alpha3.MachineDeploymen
 			Namespace:  namespace,
 		}
 		m.Log("level", "debug", "message", fmt.Sprintf("Updating infrastructure reference to  %s", machineDeployment.Name))
-		patch := mutator.PatchReplace("/spec/infrastructureRef", &infrastructureCRRef)
+		patch := mutator.PatchReplace("/spec/template/spec/infrastructureRef", &infrastructureCRRef)
 		result = append(result, patch)
 		return result, nil
 	}
 	return nil, nil
 }
 
+func (m *Mutator) MutateCAPILabel(md capiv1alpha3.MachineDeployment) []mutator.PatchOperation {
```

Comment:
Since we use this in more than one place it might make sense to move this function to the `aws` package. WDYT?

Reply:
I'm just only scratching my head how because one is taking MachineDeployment and the other one is taking Cluster

Reply:
Oh got it we can use MutateLabel

Reply:
You should be able to us `meta metav1.Object` instead.

Reply:
Ok I added still one function because of checking if empty but could still simplify a bit

Reply:
nice!